### ISSUE TITLE
v0.22.0 issue with questions depending on conversation memory when using OpenAI json response_format

### ISSUE THREAD

**Issue by qwe321389yfs8t7huqd89q** [griptape-ai/griptape#565]
- [x ] I have read and agree to the [contributing guidelines](https://github.com/griptape-ai/griptape#contributing).

**Describe the bug**
Looks like with v0.22.0 when using OpenAI `json_object` response_format most of times when you ask question that relates to previous conversation you are getting input prompt as output answer.
For few tries it worked as expected, with v0.21.2 it works for me every time.

**Additional context**
Console:
```
[01/12/24 13:48:09] INFO     ToolkitTask 25b0ed00e2ca4e67af196d168f62e788
                             Input: Output the title of post 1.
[01/12/24 13:48:18] INFO     Subtask 6798ff44b80a4267b0f5448a83479c14
                             Thought: To get the title of post 1, I need to make a GET request to the API endpoint. The
                             path parameter will be the post ID, which is 1 in this case.

                             Action:
                             {
                               "name": "RestApiClient",
                               "path": "get",
                               "input": {
                                 "values": {
                                   "path_params": [1]
                                 }
                               }
                             }
[01/12/24 13:48:39] INFO     Subtask 6798ff44b80a4267b0f5448a83479c14
                             Response: {
                               "userId": 1,
                               "id": 1,
                               "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
                               "body": "quia et suscipit\nsuscipit recusandae consequuntur expedita et
                             cum\nreprehenderit molestiae ut ut quas totam\nnostrum rerum est autem sunt rem eveniet
                             architecto"
                             }
[01/12/24 13:48:46] INFO     ToolkitTask 25b0ed00e2ca4e67af196d168f62e788
                             Output: {
                               "question": "Output the title of post 1.",
                               "intent": "Get the title of post 1 from the API.",
                               "answer": "The title of post 1 is 'sunt aut facere repellat provident occaecati excepturi
                             optio reprehenderit'."
                             }
                    INFO     ToolkitTask 25b0ed00e2ca4e67af196d168f62e788
                             Input: Now of post 2.
[01/12/24 13:48:50] INFO     ToolkitTask 25b0ed00e2ca4e67af196d168f62e788
                             Output: Now of post 2.



[01/12/24 14:20:07] INFO     ToolkitTask b4f4f1dbf72d4208b868b4435ed8942c
                             Input: Output the title of post 1.
[01/12/24 14:20:15] INFO     Subtask 3f1815da602946f7aee231f8d4a79eab
                             Thought: To get the title of post 1, I need to make a GET request to the API endpoint. The
                             post ID should be included in the path parameters of the request.

                             Action:
                             {
                               "name": "RestApiClient",
                               "path": "get",
                               "input": {
                                 "values": {
                                   "path_params": [1]
                                 }
                               }
                             }
                    INFO     Subtask 3f1815da602946f7aee231f8d4a79eab
                             Response: {
                               "userId": 1,
                               "id": 1,
                               "title": "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
                               "body": "quia et suscipit\nsuscipit recusandae consequuntur expedita et
                             cum\nreprehenderit molestiae ut ut quas totam\nnostrum rerum est autem sunt rem eveniet
                             architecto"
                             }
[01/12/24 14:20:20] INFO     ToolkitTask b4f4f1dbf72d4208b868b4435ed8942c
                             Output: {"question": "Output the title of post 1.", "intent": "GetPostTitle", "answer":
                             "sunt aut facere repellat provident occaecati excepturi optio reprehenderit"}
                    INFO     ToolkitTask b4f4f1dbf72d4208b868b4435ed8942c
                             Input: Now of post 2.
[01/12/24 14:20:24] INFO     ToolkitTask b4f4f1dbf72d4208b868b4435ed8942c
                             Output: {"question": "Now of post 2.", "intent": "GetPostTitle", "answer": "qui est esse"}
```

Memory:
from v0.22.0
```
{
  "type": "ConversationMemory",
  "runs": [
    {
      "id": "29ec4da1e47949ce8eab8ca59a10c673",
      "input": "Output the title of post 1.",
      "output": "{\n  \"question\": \"Output the title of post 1.\",\n  \"intent\": \"Get the title of post 1 from the API.\",\n  \"answer\": \"The title of post 1 is 'sunt aut facere repellat provident occaecati excepturi optio reprehenderit'.\"\n}"
    },
    {
      "id": "8584b32abe5f485a818a595e7a1989a4",
      "input": "Now of post 2.",
      "output": "Now of post 2."
    }
  ],
  "max_runs": null
}
```

from v0.21.2 and from time to time v0.22.0
```
{
  "type": "ConversationMemory",
  "runs": [
    {
      "id": "8137e223911342569063974c184a8388",
      "input": "Output the title of post 1.",
      "output": "{\"question\": \"Output the title of post 1.\", \"intent\": \"Get the title of post 1 from the API.\", \"answer\": \"The title of post 1 is 'sunt aut facere repellat provident occaecati excepturi optio reprehenderit'.\"}"
    },
    {
      "id": "784363beed6d4b2c97ae4d028aeeac5c",
      "input": "Now of post 2.",
      "output": "{\"question\": \"Now of post 2.\", \"intent\": \"Get the title of post 2 from the API.\", \"answer\": \"The title of post 2 is 'qui est esse'.\"}"
    }
  ],
  "max_runs": null
}
```

**To Reproduce**
```
import os
from dotenv import load_dotenv
from griptape.drivers import (
    AzureOpenAiChatPromptDriver,
    AzureOpenAiEmbeddingDriver,
    LocalConversationMemoryDriver,
)
from griptape.structures import Agent

from griptape.rules import Rule, Ruleset

from json import dumps
from griptape.memory.structure import ConversationMemory

from griptape.tools import RestApiClient
import string, random
from typing import Literal

load_dotenv()


def id_generator(size=12, chars=string.ascii_uppercase + string.digits):
    return "".join(random.choice(chars) for _ in range(size))


conversation_id = id_generator()

posts_client = RestApiClient(
    base_url="https://jsonplaceholder.typicode.com",
    path="posts",
    description="Allows for creating, updating, deleting, patching, and getting posts.",
    off_prompt=False,
    request_body_schema=dumps(
        {
            "$schema": "https://json-schema.org/draft/2019-09/schema",
            "$id": "http://example.com/example.json",
            "type": "object",
            "default": {},
            "title": "Root Schema",
            "required": ["title", "body", "userId"],
            "properties": {
                "title": {
                    "type": "string",
                    "default": "",
                    "title": "The title Schema",
                },
                "body": {
                    "type": "string",
                    "default": "",
                    "title": "The body Schema",
                },
                "userId": {
                    "type": "integer",
                    "default": 0,
                    "title": "The userId Schema",
                },
            },
        }
    ),
    request_query_params_schema=dumps(
        {
            "$schema": "https://json-schema.org/draft/2019-09/schema",
            "$id": "http://example.com/example.json",
            "type": "object",
            "default": {},
            "title": "Root Schema",
            "required": ["userId"],
            "properties": {
                "userId": {
                    "type": "string",
                    "default": "",
                    "title": "The userId Schema",
                },
            },
        }
    ),
    request_path_params_schema=dumps(
        {
            "$schema": "https://json-schema.org/draft/2019-09/schema",
            "$id": "http://example.com/example.json",
            "type": "array",
            "default": [],
            "title": "Root Schema",
            "items": {
                "anyOf": [
                    {
                        "type": "string",
                        "title": "Post id",
                    },
                ]
            },
        }
    ),
    response_body_schema=dumps(
        {
            "$schema": "https://json-schema.org/draft/2019-09/schema",
            "$id": "http://example.com/example.json",
            "type": "object",
            "default": {},
            "title": "Root Schema",
            "required": ["id", "title", "body", "userId"],
            "properties": {
                "id": {
                    "type": "integer",
                    "default": 0,
                    "title": "The id Schema",
                },
                "title": {
                    "type": "string",
                    "default": "",
                    "title": "The title Schema",
                },
                "body": {
                    "type": "string",
                    "default": "",
                    "title": "The body Schema",
                },
                "userId": {
                    "type": "integer",
                    "default": 0,
                    "title": "The userId Schema",
                },
            },
        }
    ),
)

output_ruleset = Ruleset(
    name="output",
    rules=[
        Rule(
            (
                "Always answer in following JSON format: ",
                """Use following JSON format: 
        {
            "question": "Put here original prompt that you received.",
            "intent": "Put here user intention",
            "answer": "Put here your answer"
        }""",
            )
        ),
    ],
)

agent = Agent(
    conversation_memory=ConversationMemory(
        driver=LocalConversationMemoryDriver(
            file_path=f"local_conversation_memory{os.sep}{conversation_id}_memory.json"
        )
    ),
    embedding_driver=AzureOpenAiEmbeddingDriver(
        api_key=os.getenv("AZURE_OPENAI_API_KEY"),
        azure_endpoint=os.getenv("AZURE_OPENAI_API_URL"),
        azure_deployment=os.getenv("AZURE_OPENAI_ADA_EMBEDDING_ID"),
        model=os.getenv("AZURE_OPENAI_ADA_EMBEDDING_MODEL"),
        api_version=os.getenv("AZURE_OPENAI_API_VERSION"),
    ),
    prompt_driver=AzureOpenAiChatPromptDriver(
        api_key=os.getenv("AZURE_OPENAI_API_KEY"),
        azure_endpoint=os.getenv("AZURE_OPENAI_API_URL"),
        azure_deployment=os.getenv("AZURE_OPENAI_GPT4_32K_ID"),
        model=os.getenv("AZURE_OPENAI_GPT4_32K_MODEL"),
        api_version=os.getenv("AZURE_OPENAI_API_VERSION"),
        response_format=Literal["json_object"],
    ),
    tools=[posts_client],
    rulesets=[output_ruleset],
)

agent_response = agent.run("Output the title of post 1.")
agent_response = agent.run("Now of post 2.")
```

**Comment by collindutter** [comment]
Thanks for the report and reproducible example @qwe321389yfs8t7huqd89q. We're looking into it and will keep you updated.

**Comment by collindutter** [comment]
@qwe321389yfs8t7huqd89q this is fixed in v0.22.1 which has just been published.
